cmd/kops/update_cluster.go

```diff
@@ -306,10 +329,17 @@ func RunUpdateCluster(ctx context.Context, f *util.Factory, clusterName string,
 		}
 		if kubecfgCert != nil {
 			klog.Infof("Exporting kubecfg for cluster")
-			conf, err := kubeconfig.BuildKubecfg(cluster, keyStore, secretStore, &commands.CloudDiscoveryStatusStore{}, clientcmd.NewDefaultPathOptions())
+			conf, err := kubeconfig.BuildKubecfg(cluster, keyStore, secretStore, &commands.CloudDiscoveryStatusStore{}, clientcmd.NewDefaultPathOptions(), c.admin)
 			if err != nil {
 				return nil, err
 			}
+
+			if c.admin {
+				conf.User = cluster.ObjectMeta.Name
+			} else {
+				conf.User = c.user
+			}
```

Comment:
I believe this logic should be pushed down into `kubeconfig.BuildKubecfg()`.

Reply:
Not sure why I did not do this earlier. Much better and the logic is now tested properly too.